elastic/beats
# Add detailed docker network summary stats

## What does this PR do?

This PR Creates a new metricset, `docker/network_summary`.  This uses network namespaces in order to fetch per-network namespace stats for a given container, similar to the counters in `system/network_summary`

I also tried to achieve some level of parity with `system/network_summary`, as it contains many of the same fields. As a result of this, we're using the same fields and same mapping methods, which have been moved to libbeat. 

## Why is it important?

A few users have been requesting data at this level, but associated with a given cgroup/container.

## Checklist


## How to test this PR locally

- pull down PR and build on a linux system with docker
- enable `docker/network_summary` in the module options.
- with some docker containers running, run this metricset, check to see if `docker.network_summary` is reporting, and that fields are populated.
